File: 2962. Count Subarrays Where Max Element Appears at Least K Times.cpp
class Solution {
public:
    long long countSubarrays(vector<int>& nums, int k) 
    {
        int n = nums.size();
        unordered_map<int,int> count;
        int maxVal = *max_element(nums.begin(), nums.end());

        long long ans = 0, i = 0, j = 0;
        while(j < n)
        {
            count[nums[j]]++;
            while(count[maxVal] == k and i <= j)
            {
                ans += n - j;
                count[nums[i++]]--;
            }
            j++;
        }
        return ans;
    }
};

File: 2962. Count Subarrays Where Max Element Appears at Least K Times.py
class Solution:
    def countSubarrays(self, nums: List[int], k: int) -> int:
        ans, n = 0, len(nums)
        maxVal = max(nums)

        i, j = 0, 0
        count = Counter()
        while j < n:
            count[nums[j]] += 1
            while count[maxVal] == k:
                count[nums[i]] -= 1
                i += 1
            ans += i
            j += 1
        return ans